### PR TITLE
feat: support set up knowhere-build-pool-size on querynode(#29650)

### PR DESCRIPTION
**Pull request by MrPresent-Han**
related: #29650

### CODE REVIEW

File: pkg/util/paramtable/component_param.go [hunk 1]
```diff
@@ -1884,12 +1884,13 @@ type queryNodeConfig struct {
...
-	InterimIndexNlist         ParamItem `refreshable:"false"`
-	InterimIndexNProbe        ParamItem `refreshable:"false"`
-	InterimIndexMemExpandRate ParamItem `refreshable:"false"`
+	KnowhereThreadPoolSize        ParamItem `refreshable:"false"`
+	ChunkRows                     ParamItem `refreshable:"false"`
+	EnableTempSegmentIndex        ParamItem `refreshable:"false"`
+	InterimIndexNlist             ParamItem `refreshable:"false"`
+	InterimIndexNProbe            ParamItem `refreshable:"false"`
+	InterimIndexMemExpandRate     ParamItem `refreshable:"false"`
+	InterimIndexBuildParallelRate ParamItem `refreshable:"true"`
```
Comment (congqixia) [inline pkg/util/paramtable/component_param.go:1893]: NIT: please rectify `refreshable` to false in next PR

**Review by congqixia — APPROVED** [review]
/lgtm

**Comment by liliu-z** [comment]
@MrPresent-Han Will this affect IndexNode?

**Comment by liliu-z** [comment]
For standalone, since the IndexNode init after the QueryNode, so it is fine. But this is to hack since the logic depends on some implicit pre-assumptions.

**Pull request merged**